RawSpeed/TiffIFD.cpp

```diff
@@ -61,6 +61,11 @@ TiffIFD::TiffIFD(FileMap* f, uint32 offset, uint32 _depth) {
       t = new TiffEntry(f, entry_offset, offset);
     } catch (IOException) { // Ignore unparsable entry
       continue;
+    } catch (...) {
+      // make sure all already allocated mEntries are freed again
+      // will be gone after cleanup using unique_ptr containers
+      this->~TiffIFD();
```

Comment:
Ah, so it was *this* easy to not leak for duplicate tags.
Though, i do not know whi